### PR TITLE
feature/restrict-the-case-access

### PR DESCRIPTION
**Pull request by AlokDatta**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-3849

### Change description ###
Restricted case access management feature will allow a caseworker to mark a sensitive case as Public, Private or Restricted. This feature is based on access management for assigned users, UI from service team and XUI - CCD enabling search for any case based on its classification. 

More details are here - https://tools.hmcts.net/confluence/display/PL/Restricted+Case+Access+Management
Changes in this PR will enable the functionality only in Sublive. There is an outstanding issue with downgrading the case security classification which will be addressed further.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/BaseCaseData.java [hunk 1]
```diff
@@ -61,4 +61,7 @@ public class BaseCaseData {
     //FPET-567 - Added for hiding fields for SDO
     @JsonProperty("isSdoSelected")
     private YesOrNo isSdoSelected;
+
+    //Amend Draft order
+    private String removeDraftOrderText;
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/models/dto/ccd/BaseCaseData.java:66]: is this change required?

File: src/main/java/uk/gov/hmcts/reform/prl/services/caseaccess/RestrictedCaseAccessService.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            }
+            default -> {
+                log.info("** inside default:: ");
+                coreCaseDataService.createCaseDataContentOnlyWithSecurityClassification(
+                    startAllTabsUpdateDataContent.startEventResponse(),
+                    Classification.PUBLIC
+                );
+            }
+        }
+        coreCaseDataService.submitUpdate(
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/caseaccess/RestrictedCaseAccessService.java:161]: For default, caseDataContent will be null, do we need to initialize it for some default value

**Review by yogendraU — APPROVED** [review]

**Review by sahaSonali — APPROVED** [review]
